cpp/ql/lib/semmle/code/cpp/UserType.qll

```diff
@@ -47,10 +47,15 @@ class UserType extends Type, Declaration, NameQualifyingElement, AccessHolder, @
     else result = this.getADeclarationLocation()
   }
 
+  private TypeDeclarationEntry getADeclarationEntryBase() {
+    type_decls(underlyingElement(result), unresolveElement(this), _)
+  }
```

Comment:
You may want to slap a `pragma[nomagic]` on this one to prevent the optimizer from inlining it and undoing your join order

Reply:
Done.